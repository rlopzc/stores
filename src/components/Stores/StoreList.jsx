import React from 'react';
import Store from './Store';
import _ from 'lodash';

const StoreList = ({ stores = [] }) => {
  if (_.isEmpty(stores)) {
    return <p>No favorite stores</p>;
  }

  return (
    <ul>
      {stores.map((store, index) => <Store key={index} {...store} />)}
    </ul>
  );
};

export default StoreList;
