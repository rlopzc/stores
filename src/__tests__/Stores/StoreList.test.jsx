import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import StoreList from '../../components/Stores/StoreList';


describe('Stores/StoreList', () => {
  const store = {position: {lat: 1, lng: 2}};
  const favorites = [store, store, store];

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<StoreList />, div);

    const component = renderer.create(<StoreList />);
    expect(component).toMatchSnapshot();
  });

  it('does not render a list when there are no favorites', () => {
    const component = shallow(<StoreList stores={[]} />);
    expect(component.find('ul').exists()).toBe(false);
  });

  it('renders a list when there are favorites', () => {
    const component = shallow(<StoreList stores={favorites} />);

    expect(component.find('ul').exists()).toBe(true);
    expect(component.find('Store').length).toBe(3);

    expect(renderer.create(<StoreList stores={favorites} />)).toMatchSnapshot();
  });
});
