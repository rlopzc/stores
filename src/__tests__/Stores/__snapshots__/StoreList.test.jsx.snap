// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stores/StoreList renders a list when there are favorites 1`] = `
<ul>
  <li />
  <li />
  <li />
</ul>
`;

exports[`Stores/StoreList renders without crashing 1`] = `
<p>
  No favorite stores
</p>
`;
